src/commands/basemaps-github/create-pr.ts

```diff
@@ -9,17 +9,56 @@ import { logger } from '../../log.js';
 import { verbose } from '../common.js';
 import { Category, MakeCogGithub, parseCategory } from './make.cog.github.js';
 
-async function parseTargetInfo(target: string): Promise<{ name: string; title: string; epsg: EpsgCode }> {
-  const splits = target.replace('s3://', '').split('/');
+const validTargetBuckets: Set<string> = new Set(['linz-basemaps', 'linz-basemaps-staging']);
+const validSourceBuckets: Set<string> = new Set(['nz-imagery', 'linz-imagery']);
+
+async function parseTargetInfo(
+  target: string,
+  individual: boolean,
+  logger: LogType,
+): Promise<{ name: string; title: string; epsg: EpsgCode; region: string | undefined }> {
+  logger.info({ target }, 'CreatePR: Get the layer information from target');
+  const url = new URL(target);
+  const bucket = url.hostname;
+  const splits = url.pathname.split('/');
   const epsg = Epsg.tryGet(Number(splits[1]));
   const name = splits[2];
+
+  //Validate the target information
+  logger.info({ bucket }, 'CreatePR: Valid the target s3 bucket');
+  if (bucket == null || !validTargetBuckets.has(bucket)) {
+    throw new Error(`Invalid s3 bucket ${bucket} from the target ${target}.`);
+  }
+
   if (epsg == null || name == null) throw new Error(`Invalid target ${target} to parse the epsg and imagery name.`);
   const collectionPath = fsa.join(target, 'collection.json');
   const collection = await fsa.readJson<StacCollection>(collectionPath);
+  if (collection == null) throw new Error(`Failed to get target collection json from ${collectionPath}.`);
   const title = collection.title;
   if (title == null) throw new Error(`Failed to get imagery title from collection.json.`);
 
-  return { name: standardizeLayerName(name), epsg: epsg.code, title };
+  //Validate the source location
+  const source = collection.links[collection.links.length - 1]?.href;
```

Comment:
you cannot trust that the last link is the correct link.

```
collection.links.find(f => f.rel == '')
```

Reply:
Very good point.